frontend/src/pages/featureStore/screens/entities/utils.ts

```diff
@@ -0,0 +1,124 @@
+import { Entity, EntityRelationship } from '#~/pages/featureStore/types/entities';
+
+/**
+ * Filters relationships for a specific entity by target type
+ * @param relationships - The relationships object from EntityList
+ * @param entityKey - The key of the entity (e.g., 'driver', 'zipcode')
+ * @param targetType - The type to filter by (e.g., 'featureService', 'featureView', 'dataSource')
+ * @returns Array of relationships that match the target type
+ */
+export const getRelationshipsByTargetType = (
+  relationships: Record<string, EntityRelationship[]>,
+  entityKey: string,
+  targetType: string,
+): EntityRelationship[] => {
+  const entityRelationships = relationships[entityKey];
+  return entityRelationships.filter((relationship) => relationship.target.type === targetType);
+};
+
+// Type guard to check if value is a record of strings
+const isRecordOfStrings = (value: unknown): value is Record<string, string> =>
+  typeof value === 'object' &&
+  value !== null &&
+  Object.values(value).every((v) => typeof v === 'string');
+
+// Type guard to check if value is a record
+const isRecord = (value: unknown): value is Record<string, unknown> =>
+  typeof value === 'object' && value !== null && !Array.isArray(value);
+
+// Helper function to get nested values from objects
+export const getNestedValue = (obj: Record<string, unknown>, path: string): unknown => {
+  const keys = path.split('.');
+  let result: unknown = obj;
+  for (const key of keys) {
+    if (result && isRecord(result) && key in result) {
+      result = result[key];
+    } else {
+      return undefined;
+    }
+  }
+  return result;
+};
+
+export const entityTableFilterKeyMapping: Record<string, string> = {
+  entity: 'spec.name',
+  joinKey: 'spec.joinKey',
+  valueType: 'spec.valueType',
+  owner: 'spec.owner',
+  project: 'project',
+  tag: 'spec.tags',
+  featureViews: 'featureViews',
+};
```

Comment:
Can we align the order in filter list w.r.t to the order in column

<img width="214" height="343" alt="Screenshot 2025-07-31 at 11 50 33 AM" src="https://github.com/user-attachments/assets/7397c52a-47a0-4e6d-b32c-e2e105020381" />